pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -0,0 +1,323 @@
+		certMapKey := keymanagementprovider.KMPMapKey{Name: keyVaultCert.Name, Version: keyVaultCert.Version}
+		certsMap[certMapKey] = certResult
+	}
+
+	return certsMap, getCertStatusMap(certsStatus), nil
+}
+
+// azure keyvault provider certificate status is a map from "certificates" key to an array of of certificate status
+func getCertStatusMap(certsStatus []map[string]string) keymanagementprovider.KeyManagementProviderStatus {
+	status := keymanagementprovider.KeyManagementProviderStatus{}
+	status[types.CertificatesStatus] = certsStatus
+	return status
+}
+
+// return a certificate status object that consist of the cert name, version and last refreshed time
+func getCertStatusProperty(certificateName, version, lastRefreshed string) map[string]string {
+	certProperty := map[string]string{}
+	certProperty[types.CertificateName] = certificateName
+	certProperty[types.CertificateVersion] = version
+	certProperty[types.CertificateLastRefreshed] = lastRefreshed
+	return certProperty
+}
+
+// formatKeyVaultCertificate formats the fields in KeyVaultCertificate
+func formatKeyVaultCertificate(object *types.KeyVaultCertificate) {
+	if object == nil {
+		return
+	}
+	objectPtr := reflect.ValueOf(object)
+	objectValue := objectPtr.Elem()
+
+	for i := 0; i < objectValue.NumField(); i++ {
+		field := objectValue.Field(i)
+		if field.Type() != reflect.TypeOf("") {
+			continue
+		}
+		str := field.Interface().(string)
+		str = strings.TrimSpace(str)
+		field.SetString(str)
+	}
+}
+
+// parseAzureEnvironment returns azure environment by name
+func parseAzureEnvironment(cloudName string) (*azure.Environment, error) {
+	var env azure.Environment
+	var err error
+	if cloudName == "" {
+		env = azure.PublicCloud
+	} else {
+		env, err = azure.EnvironmentFromName(cloudName)
+	}
+	return &env, err
+}
+
+func initializeKvClient(ctx context.Context, keyVaultEndpoint, tenantID, clientID string) (*kv.BaseClient, error) {
+	kvClient := kv.New()
+	kvEndpoint := strings.TrimSuffix(keyVaultEndpoint, "/")
+
+	err := kvClient.AddToUserAgent("ratify")
+	if err != nil {
+		return nil, re.ErrorCodeConfigInvalid.NewError(re.CertProvider, providerName, re.AKVLink, err, "failed to add user agent to keyvault client", re.PrintStackTrace)
+	}
+
+	kvClient.Authorizer, err = getAuthorizerForWorkloadIdentity(ctx, tenantID, clientID, kvEndpoint)
+	if err != nil {
+		return nil, re.ErrorCodeAuthDenied.NewError(re.CertProvider, providerName, re.AKVLink, err, "failed to get authorizer for keyvault client", re.PrintStackTrace)
+	}
+	return &kvClient, nil
+}
+
+// Parse the secret bundle and return an array of certificates
+// In a certificate chain scenario, all certificates from root to leaf will be returned
+func getCertsFromSecretBundle(ctx context.Context, secretBundle kv.SecretBundle, certName string) ([]*x509.Certificate, []map[string]string, error) {
+	if secretBundle.ContentType == nil || secretBundle.Value == nil || secretBundle.ID == nil {
```

Comment:
we have have similar logic in certStore, is the idea to have provider logic here, and eventually remove all files under certStore?